### ISSUE TITLE
Don't deploy on content-only changes

### ISSUE THREAD

**Issue by kentcdodds** [kentcdodds/remix-kentcdodds#37]
Content is read from github directly (we'll fallback to local though #36), so there's no reason to deploy the app if the only change made is in content. Thanks to help from @alexanderson1993, I made it pretty far before: https://github.com/kentcdodds/remix-kentcdodds/commit/822a1687e22986948dac519ca38fb64bf75e309c

**Comment by kentcdodds** [comment]
The approach I was following in that original attempt is probably sub-optimal. I think what would be best is to get the latest commit from the running app. Some sort of `/build-info` json endpoint that returns info on the currently running app including the commit sha of the currently deployed commit. Then we can do some git magic to determine all changes from that commit to the one we're building to determine whether anything outside `content/` changed.

**Comment by kentcdodds** [comment]
This is done. Took a completely different approach. Here's what I said in the remix discord about it:

I finally got content-only changes to not deploy and instantly refresh the cache for the content :tada: 

:rocket: Don't deploy content-only changes: https://github.com/kentcdodds/remix-kentcdodds/blob/49d4f3f37c35f25fbc5a751ebe2b21fa42b4153c/.github/workflows/deployment.yml#L188-L206
:leafy_green: Refresh content instantly: https://github.com/kentcdodds/remix-kentcdodds/blob/49d4f3f37c35f25fbc5a751ebe2b21fa42b4153c/.github/workflows/refresh-content.yml

Basically, when the site is built, I generate this json file: https://kent.dev/build/info.json during CI, I compare the files that were changed between the last deployed commit with the commit that's being built and use that to determine whether it's content-only.

Most folks don't need to bother with the non-deploy nonsense if they're using a real CMS. But I really like having my content in GitHub so people can make contributions to fix typos and stuff. This is an enormous improvement on my existing site. It took a lot more work, but it's awesome :slight_smile:
